### PR TITLE
Add updated osmFISH data that includes full FOVs.

### PR DESCRIPTION
**Pull request by ambrosejcarr**
- Add osmFISH `TileFetcher`
- Add hook for processed `osmFISH` data in `starfish.data`
- Change order of iteration over axes in builder tools (round -> ch -> z) where z is changing fastest.

### CODE REVIEW

File: starfish/experiment/builder/__init__.py [hunk 1]
```diff
@@ -89,9 +89,10 @@ def build_image(
             ImageFormat.TIFF,
         )
 
-        for z_ix in range(z_count):
-            for round_ix in range(round_count):
-                for ch_ix in range(ch_count):
+        # order reflects the most common acquisition order of the data
```
Comment (ttung) [inline starfish/experiment/builder/__init__.py]: https://github.com/spacetx/starfish/pull/758

File: examples/format_osmfish.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    @property
+    def target_map(self) -> Mapping:
+        """
+        Parse the metadata to map channel number and round to the correct target, which identifies
+        the right tile
+        """
+        parsed_metadata = {}
+        for round_, round_data in self.osmfish_metadata['HybridizationsInfos'].items():
+            round_match = re.match("Hybridization(\d{1,2})", round_)
+            if round_match is not None:
```
Comment (ttung) [inline examples/format_osmfish.py]: Do we ever expect round_match to be `None`?  If not, then we should just fatal if it's `None`.

File: examples/format_osmfish.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        self.z = z
+        self._coordinates = coordinates
+
+    @property
+    def shape(self) -> Tuple[int, ...]:
+        return self.tile_data().shape
+
+    @property
+    def coordinates(self) -> Mapping[Union[str, Coordinates], Union[Number, Tuple[Number, Number]]]:
+        """For this single field of view test experiment, we expose dummy coordinates"""
```
Comment (ttung) [inline examples/format_osmfish.py]: ?

Comment (ambrosejcarr) [reply examples/format_osmfish.py]: Oops, You can see that I was expecting to do a worse job of this PR ;-)

File: examples/format_osmfish.py (hunk 2 above)
Comment (ambrosejcarr) [reply examples/format_osmfish.py]: Yea, there are some values in the metadata that we don't have the files for, so some of these will fail. I made a note to explain this better.

File: starfish/experiment/builder/__init__.py (hunk 1 above)
Comment (ambrosejcarr) [reply starfish/experiment/builder/__init__.py]: Nice, thanks for doing this! I made a comment.

File: examples/format_osmfish.py (hunk 2 above)
Comment (ttung) [reply examples/format_osmfish.py]: Got it, understood.

I generally prefer (for loops):

```
if round_match is None:
    continue
```

Rationale:
1. reduces levels of indents
2. we know we don't need to follow the control flow past the if-statement for this case.

File: starfish/experiment/builder/__init__.py (hunk 1 above)
Comment (ttung) [reply starfish/experiment/builder/__init__.py]: Landed, you can now rebase. :)

File: starfish/data/__init__.py [hunk 4]
```diff
@@ -31,8 +31,31 @@ def ISS(use_test_data: bool=False):
...
+    See Also
+    --------
+    Codeluppi et al. 2018: https://www.nature.com/articles/s41592-018-0175-z
+    """
+    if use_test_data:
+        return Experiment.from_json(
+            'https://d2nhj9g34unfro.cloudfront.net/20181005/osmFISH/experiment.json')
     return Experiment.from_json(
-        'https://d2nhj9g34unfro.cloudfront.net/20181005/osmFISH/experiment.json')
+        'https://d2nhj9g34unfro.cloudfront.net/20181031/osmFISH/experiment.json')
```
Comment (ttung) [inline starfish/data/__init__.py]:
```suggestion
        "https://d2nhj9g34unfro.cloudfront.net/20181031/osmFISH/experiment.json")
```

File: examples/format_osmfish.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+          (channel, round) tuple. The files do not contain channel information,
+        - The spatial organization of the fields of view are not known to the starfish developers,
+          so they are filled by dummy coordinates
+        - This TileFetcher is specifically tailored to the gene panel used for a specific
+          experiment. Generalization of this TileFetcher will require reimplementation of the
+          `channel_map` method.
+        """
+        import yaml
+
+        with open(metadata_yaml, 'r') as f:
```
Comment (ttung) [inline examples/format_osmfish.py]:
```suggestion
        with open(metadata_yaml, "r") as f:
```

File: examples/format_osmfish.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        y_pixels, x_pixels, z_pixels = (
+            self.osmfish_metadata['ImageProperties']['HybImageSize'].values()
+        )
+        y_size = y_pixels * pixel_size
+        x_size = x_pixels * pixel_size
+        z_size = z_pixels * pixel_size
+        position_string = (
+            self.osmfish_metadata['TilesPositions'][f'Hybridization{round_ + 1}'][z]
+        )
+        y_pos, x_pos, z_pos = (float(v) for v in position_string.split(', '))
```
Comment (ttung) [inline examples/format_osmfish.py]:
```suggestion
        y_pos, x_pos, z_pos = (float(v) for v in position_string.split(", "))
```

File: examples/format_osmfish.py [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        Indices.ROUND: 13,
+        Indices.CH: len(primary_tile_fetcher.channel_map),
+        Indices.Z: primary_tile_fetcher.num_z
+    }
+
+    def postprocess_func(experiment_json_doc):
+        experiment_json_doc["codebook"] = "codebook.json"
+        return experiment_json_doc
+
+    with open(os.path.join(output_dir, "codebook.json"), 'w') as f:
```
Comment (ttung) [inline examples/format_osmfish.py]:
```suggestion
    with open(os.path.join(output_dir, "codebook.json"), "w") as f:
```

**Comment by joshmoore** [comment]
Travis master is unhappy. See gh-784.

**Pull request merged**